src/components/puck/Promo.tsx

```diff
@@ -0,0 +1,340 @@
+import * as React from "react";
+import { ComponentConfig, Fields } from "@measured/puck";
+import { Image, ImageProps, ImageType } from "@yext/pages-components";
+import {
+  yextCn,
+  useDocument,
+  resolveYextEntityField,
+  EntityField,
+  YextEntityField,
+  YextEntityFieldSelector,
+  NumberOrDefault,
+  HeadingProps,
+  NumberFieldWithDefaultOption,
+  CTAProps,
+  ImageWrapperProps,
+  Section,
+  Heading,
+  Body,
+  CTA,
+} from "../../index.js";
+import { imageWrapperVariants } from "./Image.tsx";
```

Comment:
Do .js imports in the components because the starter has a more basic tsconfig that doesn't allow ts imports